examples/reference/custom_components/Viewer.md

```diff
@@ -0,0 +1,258 @@
+- **`servable`**: This method serves the component using Panel's built-in server when running `panel serve ...`.
+- **`show`**: Displays the component in a new browser tab when running `python ...`.
+
+## Usage
+
+### Styling with CSS
+
+You can style the component by styling the component(s) returned by `__panel__` using their `styles` or `stylesheets` attributes.
+
+```pyodide
+import panel as pn
+import param
+
+from panel.viewable import Viewer
+
+pn.extension()
+
+
+class StyledCounterButton(Viewer):
+
+    value = param.Integer()
+
+    _stylesheets = [
+        """
+        :host(.solid) .bk-btn.bk-btn-default
+        {
+            background: #0072B5;
+            color: white;
+            border: none;
+            padding: 10px;
+            border-radius: 4px;
+        }
+        :host(.solid) .bk-btn.bk-btn-default:hover {
+            background: #4099da;
+        }
+        """
+    ]
+
+    def __init__(self, **params):
+        super().__init__()
+
+        self._layout = pn.widgets.Button(
+            name=self._button_name,
+            on_click=self._on_click,
+            stylesheets=self._stylesheets,
+            **params,
+        )
+
+    def _on_click(self, event):
+        self.value += 1
+
+    @param.depends("value")
+    def _button_name(self):
+        return f"Clicked {self.value} times"
+
+    def __panel__(self):
+        return self._layout
+
+
+StyledCounterButton().servable()
+```
+
+See the [Apply CSS](../../how_to/styling/apply_css.md) guide for more information on styling Panel components.
+
+## Displaying A Single Child
+
+You can display Panel components (`Viewable`s) by defining a `Child` parameter.
+
+Let's start with the simplest example:
+
+```pyodide
+import panel as pn
+
+from panel.custom import Child
+from panel.viewable import Viewer
+
+class SingleChild(Viewer):
+
+    object = Child()
+
+    def __panel__(self):
+      return pn.Column("A Single Child", self._object)
+
+    @pn.depends("object")
+    def _object(self):
+      return self.object
+
+single_child = SingleChild(object=pn.pane.Markdown("A **Markdown** pane!"))
+single_child.servable()
+```
+
+The `_object` is a workaround to enable the `_layout` to replace the `object` component dynamically.
+
+Let's replace the `object` with a `Button`:
+
+```pyodide
+single_child.object = pn.widgets.Button(name="Click me")
+```
+
+Let's change it back
+
+```pyodide
+single_child.object = pn.pane.Markdown("A **Markdown** pane!")
+```
+
+If you provide a non-`Viewable` child it will automatically be converted to a `Viewable` by `pn.panel`:
+
+```pyodide
+SingleChild(object="A **Markdown** pane!").servable()
+```
+
+If you want to allow a certain type of Panel components only, you can specify the specific type in the `class_` argument.
+
+```pyodide
+import panel as pn
+
+from panel.custom import Child
+from panel.viewable import Viewer
+
+class SingleChild(Viewer):
+
+    object = Child(class_=pn.pane.Markdown)
+
+    def __panel__(self):
```

Comment:
Would so nice to get rid of `def _object`. But I cannot get it working without a work around like this: `self.param.object` does not display it in the way I want.

Reply:
This should do it

```python
pn.Column("A Single Child", pn.param.ParamRef(self.param.object))
```

Reply:
Hmm. That is nice to know. But how would anyone but you know? I've never seen `ParamRef` mentioned or used before.

Reply:
It's a pretty recent addition and may need to be documented better (or even renamed). Originally it was introduced as a baseclass but it is useful when you want to render valid param references (parameter objects, generators, `rx` etc.).